src/deterministic.jl

```diff
@@ -18,26 +18,64 @@ function Deterministic(
     data::Union{Dict{Dates.DateTime, Any}, SortedDict{Dates.DateTime, Any}};
     normalization_factor::NormalizationFactor = 1.0,
     scaling_factor_multiplier::Union{Nothing, Function} = nothing,
-    timestamp = :timestamp,
     resolution::Union{Dates.Period, Nothing} = nothing,
 )
     for (k, v) in data
         if v isa DataFrames.DataFrame
-            data[k] = TimeSeries.TimeArray(v; timestamp = timestamp)
+            data[k] = Array(v)
         elseif v isa TimeSeries.TimeArray
-            continue
+            data[k] = TimeSeries.values(v)
+            if resolution === nothing
+                resolution = diff(TimeSeries.timestamp(v))[1]
```

Comment:
This computes a diff of the entire timestamp array and then takes the first element.  You just need to subtract the first two elements.